azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -58,10 +58,6 @@
   public ProcessJob(final String jobId, final Props sysProps,
       final Props jobProps, final Logger log) {
     super(jobId, sysProps, jobProps, log);
-
-    // this is in line with what other job types (hadoopJava, spark, pig, hive)
-    // is doing
-    jobProps.put(CommonJobProperties.JOB_ID, jobId);
```

Comment:
In order to use ${azkaban.job.id} in job file or common.properties file, we need to resolve this variable when building job types. This happens earlier than configuring the JOB_ID in the jobProps. So at the time when it's building job types, it cannot find JOB_ID.